components/dashboard/src/Menu.tsx

```diff
@@ -58,6 +58,14 @@ export default function Menu() {
     const showTeamsUI = user?.rolesOrPermissions?.includes('teams-and-projects');
     const team = getCurrentTeam(location, teams);
 
+    if (team) {
+        // updating last team selection
+        try {
+            localStorage.setItem('team-selection', team.slug);
```

Comment:
And how can you revert back to User is selected by default? Current code reads like if you select a team once, you'll forever have a team selected by default, never your own user.

Also, I find restoring this selection on page load quite dangerous, and would personally prefer we rely on URLs for correct team selection (i.e. no local storage + restore on root).